Client/Services/NotificationService.cs

```diff
@@ -0,0 +1,79 @@
+using BlazorApp.Shared;
+using Microsoft.JSInterop;
+using System.Threading.Tasks;
+
+namespace BlazorApp.Client.Services
+{
+    public class NotificationService
+    {
+        private readonly IJSRuntime _jsRuntime;
+
+        public NotificationService(IJSRuntime jsRuntime)
+        {
+            _jsRuntime = jsRuntime;
+        }
+
+        /// <summary>
+        /// Checks if notifications are supported in the browser
+        /// </summary>
+        public async Task<bool> IsNotificationSupported()
+        {
+            return await _jsRuntime.InvokeAsync<bool>("notifications.isSupported");
+        }
+
+        /// <summary>
+        /// Gets the current notification permission status
+        /// </summary>
+        /// <returns>Permission status: "granted", "denied", "default", or "unsupported"</returns>
+        public async Task<string> GetPermissionStatus()
+        {
+            return await _jsRuntime.InvokeAsync<string>("notifications.getPermissionStatus");
+        }
+
+        /// <summary>
+        /// Requests notification permission from the user
+        /// </summary>
+        /// <returns>Permission status: "granted", "denied", "default", or "unsupported"</returns>
+        public async Task<string> RequestPermission()
+        {
+            return await _jsRuntime.InvokeAsync<string>("notifications.requestPermission");
+        }
+
+        /// <summary>
+        /// Registers a periodic background sync for medication reminders
+        /// </summary>
+        public async Task<bool> RegisterPeriodicSync(TrackedItem item)
+        {
+            // Check if the item has any scheduled targets
+            var eligibleTargets = item.Targets.Where(t => t.Qty > 0 && t.Frequency > TimeSpan.Zero).ToList();
```

Comment:
We can just use the `GetFutureOccurrences` method from `TrackedItem` to find out when the next instance is. As long as it's in the future, that's how long we know we need to wait.